examples/vrf/src/handlers/wire.rs

```diff
@@ -0,0 +1,382 @@
+use bytes::{Buf, BufMut};
+use commonware_codec::{EncodeSize, Error, FixedSize, Read, ReadExt, ReadRangeExt, Write};
+use commonware_cryptography::bls12381::primitives::{
+    group,
+    poly::{self, Eval},
+};
+use commonware_utils::{quorum, Array};
+use std::collections::HashMap;
+
+/// Represents a top-level message for the Distributed Key Generation (DKG) protocol,
+/// typically sent over a dedicated DKG communication channel.
+///
+/// It encapsulates a specific round number and a payload containing the actual
+/// DKG protocol message content.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Dkg<Sig: Array> {
+    pub round: u64,
+    pub payload: Payload<Sig>,
+}
+
+impl<Sig: Array> Write for Dkg<Sig> {
+    fn write(&self, buf: &mut impl BufMut) {
+        self.round.write(buf);
+        self.payload.write(buf);
+    }
+}
+
+impl<Sig: Array> Read<usize> for Dkg<Sig> {
+    fn read_cfg(buf: &mut impl Buf, num_players: &usize) -> Result<Self, Error> {
+        let round = u64::read(buf)?;
+        let payload = Payload::<Sig>::read_cfg(buf, num_players)?;
+        Ok(Self { round, payload })
+    }
+}
+
+impl<Sig: Array> EncodeSize for Dkg<Sig> {
+    fn encode_size(&self) -> usize {
+        self.round.encode_size() + self.payload.encode_size()
+    }
+}
+
+/// Defines the different types of messages exchanged during the DKG protocol.
+///
+/// This enum is used as the `payload` field within the [`Dkg`] message struct.
+/// The generic parameter `Sig` represents the type used for signatures in acknowledgments.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Payload<Sig: Array> {
```

Comment:
If you end up embedding `verify`/`sign` on the type, you'll need to move these to standalone structs (I think)>